### PR TITLE
Add gcp secret parameter to container op

### PR DESCRIPTION
**Pull request by IronPan**
This would allow specifying gcp secret to the container, for accessing all GCP API.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/261)

### CODE REVIEW

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -144,6 +144,17 @@ def _op_to_template(self, op):
       if op.cpu_request:
         template['container']['resources']['requests']['cpu'] = op.cpu_request
 
+    if op.gcp_secret:
+      template['container']['env'] = {}
```
Comment (Ark-kun) [inline sdk/python/kfp/compiler/compiler.py]: template['container']['env'] = {
        'env': {
          'name': 'GOOGLE_APPLICATION_CREDENTIALS',
          'value': '/secret/gcp-credentials/user-gcp-sa.json',
        }
      }
      template['container']['volumeMounts'] = {
        'mountPath': '/secret/gcp-credentials',
        'name': 'gcp-credentials',
      }
      template['volumes'] = {
        'name': 'gcp-credentials',
        'secret': {
          'secretName': op.gcp_secret,
        }
      }

File: sdk/python/kfp/compiler/compiler.py [hunk 2]
```diff
@@ -144,6 +144,17 @@ def _op_to_template(self, op):
...
+      template['container']['env']['name'] = 'GOOGLE_APPLICATION_CREDENTIALS'
+      template['container']['env']['value'] = '/secret/gcp-credentials/user-gcp-sa.json'
+      template['container']['volumeMounts'] = {}
+      template['container']['volumeMounts']['mountPath'] = '/secret/gcp-credentials'
+      template['container']['volumeMounts']['name'] = 'gcp-credentials'
+      template['volumes'] = {}
```
Comment (Ark-kun) [inline sdk/python/kfp/compiler/compiler.py]: Volumes and volume mounts should be arrays.

File: sdk/python/tests/compiler/compiler_tests.py [hunk 3]
```diff
@@ -35,14 +35,22 @@ def test_operator_to_template(self):
...
-                           file_outputs={'merged': '/tmp/message.txt'})
+                           file_outputs={'merged': '/tmp/message.txt'}, gcp_secret='user-gcp-sa')
     golden_output = {
       'container': {
         'image': 'image',
         'args': [
           'echo {{inputs.parameters.msg1}} {{inputs.parameters.msg2}} | tee /tmp/message.txt'
         ],
         'command': ['sh', '-c'],
+        'env': {
```
Comment (Ark-kun) [inline sdk/python/tests/compiler/compiler_tests.py]: Did you test running this pipeline?

Comment (Ark-kun) [reply sdk/python/tests/compiler/compiler_tests.py]: Please create the secret in your cluster, compile a pipeline, submit it to the cluster and check for errors.

**Comment by Ark-kun** [comment]
It would be great to have an integration test for this in future.

File: sdk/python/tests/compiler/compiler_tests.py (hunk 3 above)
Comment (gaoning777) [reply sdk/python/tests/compiler/compiler_tests.py]: Agree with Alexey for testing it manually before merging

**Comment by Ark-kun** [comment]
Added issue about per-container volumes to Argo https://github.com/argoproj/argo/issues/1094

**Pull request merged**